src/config.ts

```diff
@@ -139,22 +166,24 @@ function readConfigFile(path: string | undefined): ConfigFile | undefined {
     throw exn;
   }
 
-  let config;
-  try {
-    config = YAML.parse(fileContent) as unknown;
-  } catch (exn) {
-    assert(isNativeError(exn));
+  const document = YAML.parseDocument(fileContent, { keepSourceTokens: true });
+  if (document.errors.length > 0) {
+    const errorMessage = document.errors.map((e) => `${e.name}: ${e.message}`).join("\n");
     throw new Error(
-      `Error parsing config file at ${path}: ${exn.message}\nYou can remove the file to use the default configuration.`,
+      `Failed to parse config file at ${path}\n${errorMessage}\nYou can remove the file to use the default configuration.`,
     );
   }
-  if (!config) return;
+  return document;
+}
 
+function readConfigFile(document: Document): ConfigFile {
+  const config = document.toJSON() as Record<string, unknown>;
```

Comment:
Please don't do a forced coercion like that. Instead, coerce to `unknown` and let `assert`s below carve the actual type.

Reply:
To be fair, TypeScript doesn't provide great options for this situation. When loading data from a file, there is not much alternative to type coercion.

Reply:
That's not true. What I suggested is literally how to do it and it works really well.

Reply:
OK but I mean, so what? If the file is somehow mangled or corrupted, it's still going to error.